### ISSUE TITLE
Dashboard shows only a blank page in Safari

### ISSUE THREAD

**Issue by phantomski77** [emqx/emqx#11822]
### What happened?

Dashboard used to open without problem. Suddenly in Safari it only shows a blank page. Same dashboard opens correctly in Firefox and Edge, even in the same Safari private browsing. Reloading the page or erasing cache doesn't help. Restarting or spinning a new container doesn't help. The client is not banned in banned clients list. Warning msg in the container log. No modifications to config files, just basic example setup.

### What did you expect to happen?

Dashboard to open normally.

### How can we reproduce it (as minimally and precisely as possible)?

- docker 24.0.6 on the server (arm64 architecture Ubuntu 22.04.3 LTS), standard bridge networking
- official emqx container emqx/emqx:5.3.0 with basic example setup, http only, dashboard on port 18083
- MacOS Sonoma 14.0 on Apple MacBook Pro M2 Max using Safari 17.0 (19616.1.27.211.1) as a client

### EMQX version

<details>

```console
$ ./bin/emqx_ctl broker
sysdescr  : EMQX
version   : 5.3.0
datetime  : 2023-10-26T11:50:49.950076598+00:00
uptime    : 1 hours, 17 minutes, 31 seconds
```

</details>

### OS version

<details>

```console
$ cat /etc/os-release
PRETTY_NAME="Ubuntu 22.04.3 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.3 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy

$ uname -a
Linux <hostname> 5.15.0-1041-raspi #44-Ubuntu SMP PREEMPT Tue Oct 10 13:05:59 UTC 2023 aarch64 aarch64 aarch64 GNU/Linux

```

</details>

### Log files

<details>

```log
2023-10-26T10:37:28.527283+00:00 [warning] msg: unexpected_api_access, mfa: emqx_dashboard_not_found:init/2(25), request: #{bindings => #{},body_length => 0,cert => undefined,has_body => false,headers => #{<<"accept">> => <<"*/*">>,<<"accept-encoding">> => <<"gzip, deflate">>,<<"accept-language">> => <<"en-GB,en;q=0.9">>,<<"connection">> => <<"keep-alive">>,<<"host">> => <<"[hostname]:18083">>,<<"user-agent">> => <<"com.apple.WebKit.Networking/19616.1.27.211.1 CFNetwork/1474.1 Darwin/23.0.0">>},host => <<"[hostname]">>,host_info => undefined,method => <<"GET">>,path => <<"/apple-touch-icon.png">>,path_info => undefined,peer => {{[client_ip]},54561},pid => <0.3579.0>,port => 18083,qs => <<>>,ref => 'http:dashboard',scheme => <<"http">>,sock => {{[container_ip]},18083},streamid => 1,version => 'HTTP/1.1'}
```
</details>

**Comment by zmstone** [comment]
the warning log seems to be normal, because the browser is trying to get the icon `/apple-touch-icon.png` which is not provided by emqx dashboard web server.

I cannot seem to reproduce this issue in my macbook.
could you try to see if you can find something from the developers console ?
https://support.apple.com/guide/safari/use-the-developer-tools-in-the-develop-menu-sfri20948/mac

**Comment by phantomski77** [comment]
> could you try to see if you can find something from the developers console ?

I've tried that right away. It basically returns code 431
This is the return code:
```http
Failed to load resource: the server responded with a status of 431 (Request Header Fields Too Large)
```
But all I'm requesting is the Dashboard, nothing is parsed and the link is exactly the same that opens elsewhere. Even in the same Safari, just it the Privacy Browsing tab. I couldn't find anything stored, no cookies, even deleted the cache.

**Comment by id** [comment]
It's weird that it works fine in Privacy Browsing tab.
Could you double check that you don't have any cookies for this site?

In JavaScript console:
```
console.log(document.cookie);
```

**Comment by phantomski77** [comment]
> Could you double check that you don't have any cookies for this site?

Hmmm, now that's interesting. Actually there were some, from other services running on the same site.

If I delete all cookies, EMQX Dashboard starts fine.

I'm running few more services as docker containers on different ports at the same server. Most of them are fine. When I've opened [n8n's](https://n8n.io) dashboard  (workflow automation tool, v1.11.1 on port 5678) however, the cookies this service stores immediately tripped out EMQX which responds with server error in the Dashboard and on page reload comes up with code 431.

Strange thing is, it doesn't always happen. It happened when I've deleted them all at first, but not on the second attempt. It's hard to replicate, happens out of the blue. As far as I can see, n8n is using RudderEncrypt for tracking, but hard to see if it's related.
